### ISSUE TITLE
Failed to load the jupyterlab-git server extension

### ISSUE THREAD

**Issue by maxbernyk** [jupyterlab/jupyterlab-git#1202]
jupyterlab-git extension fails to load in the latest `JupyterLab v3.5.1` (last working version is v3.4.8). Tested on Linux and Mac OS.

`RuntimeError: Calling 'GitSettingsHandler.get_current_user()' directly is deprecated in jupyter-server 2.0. Use 'self.current_user' instead (works in all versions).`

```
$ jupyter labextension list

JupyterLab v3.5.1
/home/xxx/.venv/share/jupyter/labextensions
        jupyterlab_pygments v0.2.2 enabled OK (python, jupyterlab_pygments)
        nbdime-jupyterlab v2.1.1 enabled OK
        @jupyterlab/git v0.40.1 enabled OK (python, jupyterlab-git)
```
![Screenshot_20221209_124455](https://user-images.githubusercontent.com/4667838/206605235-1740d8e4-673f-4c86-86bd-7179ae63c2d1.png)

**Comment by tarrade** [comment]
I had the same issue and seems to be related to the version 2.0 of jupyter_server. Try to use `jupyter_server==1.23.3`. For me I can use it with` jupyterlab-git==0.40.1`

**Comment by fcollonval** [comment]
Thanks for reporting @maxbernyk 

Would you mind providing the full error stack trace (it will ease figure out what is wrong)?

**Comment by maxbernyk** [comment]
```
[I 2022-12-12 21:51:12.982 ServerApp] jupyter_server_mathjax | extension was successfully linked.
[I 2022-12-12 21:51:12.989 ServerApp] jupyter_server_terminals | extension was successfully linked.
[I 2022-12-12 21:51:12.998 ServerApp] jupyterlab | extension was successfully linked.
[I 2022-12-12 21:51:12.999 ServerApp] jupyterlab_git | extension was successfully linked.
[W 2022-12-12 21:51:13.002 NotebookApp] 'token' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2022-12-12 21:51:13.006 ServerApp] ServerApp.token config is deprecated in 2.0. Use IdentityProvider.token.
[I 2022-12-12 21:51:13.007 ServerApp] nbclassic | extension was successfully linked.
[I 2022-12-12 21:51:13.007 ServerApp] nbdime | extension was successfully linked.
[I 2022-12-12 21:51:13.322 ServerApp] notebook_shim | extension was successfully linked.
[W 2022-12-12 21:51:13.628 ServerApp] All authentication is disabled.  Anyone who can connect to this server will be able to run code.
[I 2022-12-12 21:51:13.631 ServerApp] notebook_shim | extension was successfully loaded.
[I 2022-12-12 21:51:13.634 ServerApp] jupyter_server_mathjax | extension was successfully loaded.
[I 2022-12-12 21:51:13.638 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2022-12-12 21:51:13.641 LabApp] JupyterLab extension loaded from /home/xxx/.venv/lib/python3.9/site-packages/jupyterlab
[I 2022-12-12 21:51:13.641 LabApp] JupyterLab application directory is /home/xxx/.venv/share/jupyter/lab
[I 2022-12-12 21:51:13.650 ServerApp] jupyterlab | extension was successfully loaded.
[I 2022-12-12 21:51:13.660 ServerApp] jupyterlab_git | extension was successfully loaded.
[I 2022-12-12 21:51:13.668 ServerApp] nbclassic | extension was successfully loaded.
[I 2022-12-12 21:51:13.783 ServerApp] nbdime | extension was successfully loaded.
[I 2022-12-12 21:51:13.784 ServerApp] Serving notebooks from local directory: /home/xxx
[I 2022-12-12 21:51:13.785 ServerApp] Jupyter Server 2.0.1 is running at:
[I 2022-12-12 21:51:13.785 ServerApp] http://localhost:8888/lab
[I 2022-12-12 21:51:13.785 ServerApp]  or http://127.0.0.1:8888/lab
[I 2022-12-12 21:51:13.785 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[I 2022-12-12 21:51:18.947 LabApp] Generating new user for token-authenticated request: a6d95c857acf4b56810b9f5c3f3651f1
[W 2022-12-12 21:51:22.821 LabApp] Could not determine jupyterlab build status without nodejs
/home/xxx/.venv/lib/python3.9/site-packages/jupyterlab_git/handlers.py:43: RuntimeWarning: coroutine 'APIHandler.prepare' was never awaited
  super().prepare()
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
[E 2022-12-12 21:51:23.544 ServerApp] Uncaught exception GET /git/settings?version=0.40.1&1670842283427 (127.0.0.1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/git/settings?version=0.40.1&1670842283427', version='HTTP/1.1', remote_ip='127.0.0.1')
    Traceback (most recent call last):
      File "/home/xxx/.venv/lib/python3.9/site-packages/tornado/web.py", line 1711, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/home/xxx/.venv/lib/python3.9/site-packages/tornado/web.py", line 3194, in wrapper
        if not self.current_user:
      File "/home/xxx/.venv/lib/python3.9/site-packages/tornado/web.py", line 1345, in current_user
        self._current_user = self.get_current_user()
      File "/home/xxx/.venv/lib/python3.9/site-packages/jupyter_server/base/handlers.py", line 158, in get_current_user
        raise RuntimeError(msg)
    RuntimeError: Calling `GitSettingsHandler.get_current_user()` directly is deprecated in jupyter-server 2.0. Use `self.current_user` instead (works in all versions).
[W 2022-12-12 21:51:23.554 ServerApp] wrote error: 'Unhandled error'
[E 2022-12-12 21:51:23.556 ServerApp] {
      "Host": "localhost:8888",
      "Accept": "*/*",
      "Referer": "http://localhost:8888/lab",
      "User-Agent": "Mozilla/5.0 (X11; Linux aarch64; rv:102.0) Gecko/20100101 Firefox/102.0"
    }
[E 2022-12-12 21:51:23.556 ServerApp] 500 GET /git/settings?version=0.40.1&1670842283427 (@127.0.0.1) 27.19ms referer=http://localhost:8888/lab
```

**Comment by fcollonval** [comment]
@maxbernyk I released version 0.41.0 that should be compatible with Jupyter Server 2. Let me know if you are still seeing some errors.

**Comment by maxbernyk** [comment]
works now with the latest jupyter-server - thanks!
